### DISCUSSION TITLE
Script for running flutter pub get in all the folders

### DISCUSSION THREAD

**Discussion by shan-shaji** [serverpod/serverpod#605] [Ideas]
Would be nice if we have VS Code task that will enable us to run the `flutter pub get` in all the folders.

**Comment by yahu1031** [comment]
@shan-shaji Could you please elaborate on it? AOMU you are talking about any vs code extension(serverpod)?

**Comment by shan-shaji** [comment]
@yahu1031 i was talking about a build script that will run ` flutter pub get` in all of the dart/folder folders. Since serverpod have multiple folders which contains dart/flutter projects it will be tedious to go to each folder and run `flutter pub get` during the initial setup.

**Reply by yahu1031** [reply]
That could be possible.

**Reply by II11II** [reply]
Are you looking inside serverpod codebase or just developing backend?
If you are looking inside serverpod and you need to get all dependencies  you can run existence shell comand inside serverpod project
But if you are just trying serverpod for your backend, i think you you can get all dependencies by running in 3 folders or you can create bash script (see scripts inside serverpod's source code)
Or there is also another tool which called **melos**  you can find it on pub.dev

**Reply by shan-shaji** [reply]
I was looking inside the serverpod codebase. Found it thank you. What i was suggesting is to create tasks in VScode by using these scripts.

**Reply by yahu1031** [reply]
@shan-shaji We better don't do it(preferred). Cause we are not opening the vscode after the project was created. And to do it(pub get) in vscode, we need an extension that will prompt that. Where the flutter extension will do it by default.

**Reply by shan-shaji** [reply]
i am talking about the serverpod codebase. Say i want to contribute to serverpod after cloning i had to visit every folder and run `pub get` on each folder.  So i thought if we had a VS code task it would have been easy and i was not aware about the existing scripts.

**Reply by yahu1031** [reply]
Even this will be handled by the vscode itself. Cause any dart/flutter code identified by the extension, it will do for all the pub gets if you just click on pub get.

**Reply by shan-shaji** [reply]
@yahu1031  Cool,  but i believe if we are opening the entire serverpod code base in vs code we won't get that option. I think that is the reason why we have a script to run pub get in all dart and flutter folders.

**Reply by shan-shaji** [reply]
![Uploading image.png…]()

**Reply by yahu1031** [reply]
Image missing, but yeah it will do by default. That will be taken care of by the flutter extension itself(serverpod created project). And moreover you don't need to open the whole serverpod project (I assume you are talking about this repo) to make some changes.

**Reply by shan-shaji** [reply]
yup

**Reply by shan-shaji** [reply]
i was talking about this repo

**Reply by fischerscode** [reply]
Did you try `util/pub_get_all`?

**Reply by shan-shaji** [reply]
@fischerscode yes. I was talking about adding that script to create a Vscode task. nvm i don't think it's a thing that needs to be focused.

**Reply by wstrange** [reply]
This would be a good fit for `melos`

**Comment by shan-shaji** [comment]
@yahu1031 Could you please tell me how to fix this error

`SERVERPOD_HOME environmental variable is required in development mode`

Got it when i tried to run serverpod after installing serverpod_cli from master.

**Reply by II11II** [reply]
@shan-shaji 
run this command: ```dart pub global activate serverpod_cli```

**Reply by shan-shaji** [reply]
This command will install the package directly from pub.dev right? I want to test a change locally inside the CLI.

**Reply by yahu1031** [reply]
@shan-shaji You need to set `SERVERPOD_HOME` variable in the env variables.

Or alternatively, override the code block to `true`***(NOT RECOMMENDED)***

**Reply by shan-shaji** [reply]
@yahu1031 what should the be the value?

**Reply by yahu1031** [reply]
@shan-shaji What I do is, I created a `.serverpod` folder at `$HOME`  and I included this in the path

```sh
export SERVERPOD_HOME='$HOME/.serverpod'
export PATH='$PATH:$SERVERPOD_HOME'
```

**Reply by shan-shaji** [reply]
ok cool, Thanks @yahu1031

**Reply by vlidholt** [reply]
SERVERPOD_HOME should point to to root directory of your cloned serverpod repository.
